homeassistant/components/broadlink/switch.py

```diff
@@ -54,12 +68,55 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     mac_addr = config.get(CONF_MAC)
     switches = config.get(CONF_SWITCHES)
 
+    if not isinstance(switches, list):
+        switches = [
+            {
+                CONF_NAME: switch[CONF_FRIENDLY_NAME],
+                CONF_COMMAND_OFF: switch[CONF_COMMAND_OFF],
+                CONF_COMMAND_ON: switch[CONF_COMMAND_ON],
+            }
+            for switch in switches.values()
+        ]
+
+        _LOGGER.warning(
+            "Your configuration for the switch platform is deprecated. "
+            "Please refer to the Broadlink documentation to catch up"
+        )
+
     if mac_addr and switches:
+        for attr in {CONF_TYPE, CONF_TIMEOUT}:
+            if config.get(attr) is not None:
+                _LOGGER.warning(
```

Comment:
We have a validator `cv.deprecated` that we can use.

Reply:
Much better. Thank you.